### PR TITLE
[Draft] NYX Eiger2 Flyer initial commit

### PR DESCRIPTION
**Pull request by mskinner5278**
Still not sure why the history switches between the commits, I may just not restore the old flyer in order to preserve the history and manually copy/add it back in.

### CODE REVIEW

File: nyxtools/flyer-eiger2.py [hunk 1]
```diff
@@ -16,37 +16,15 @@
...
+class NYXEiger2Flyer(MXFlyer):
     def __init__(self, vector, zebra, detector=None) -> None:
-        self.name = "NYXFlyer"
-        self.vector = vector
-        self.zebra = zebra
-        self.detector = detector
-
+        self.super().__init__(vector, zebra, detector)
+        self.name = "NYXEiger2Flyer"
         self.data_directory_name = None
```
Comment (JunAishima) [inline nyxtools/flyer-eiger2.py:17]: do you really need to keep this variable in?

File: nyxtools/flyer-eiger2.py [hunk 2]
```diff
@@ -16,37 +16,15 @@
...
-        self.file_prefix = None
-        self.num_images = None
-        self.file_number_start = None
-
-        self._asset_docs_cache = deque()
         self._resource_document = None
```
Comment (JunAishima) [inline nyxtools/flyer-eiger2.py:18]: this shouldn't be necessary at all - it's only used in the Pilatus-specific collect_asset_docs()

File: nyxtools/flyer-eiger2.py [hunk 3]
```diff
@@ -16,37 +16,15 @@
...
-        # self._datum_counter = None
-        # TODO: rework to use datum ids dictionary.
-        # self._datum_ids = DEFAULT_DATUM_DICT
-        self._datum_ids = []
-        self._first_file = None
-
-        self._collection_dictionary = None
 
     def update_parameters(self, **kwargs):
         self.data_directory_name = kwargs.get("data_directory_name", "/nyx-data/test")
```
Comment (JunAishima) [inline nyxtools/flyer-eiger2.py:21]: this is already done in MXTools - is there any reason we need to use this?

**Review by JunAishima — CHANGES_REQUESTED** [review]

File: nyxtools/flyer-eiger2.py (hunk 1 above)
Comment (mskinner5278) [reply nyxtools/flyer-eiger2.py:17]: I was not sure, so I decided to leave it. We could remove it and see if any failures occur due to it being missing, but I our time-line was originally going to be very short so I thought getting it functional quickly would be more important.

File: nyxtools/flyer-eiger2.py (hunk 2 above)
Comment (mskinner5278) [reply nyxtools/flyer-eiger2.py:17]: I will be sure to get rid of that one.

**Comment by mskinner5278** [comment]
I have to re-post this PR to see if it fixes the missing history.

**Pull request closed without merging**